python/lsst/ts/tcpip/base_client_or_server.py

```diff
@@ -388,6 +443,34 @@ async def write_from(self, *structs: ctypes.Structure) -> None:
         assert self.writer is not None  # make mypy happy
         await utils.write_from(self.writer, *structs)
 
+    async def write_str(self, line: str) -> None:
+        """Write a string of data.
+
+        Before writing, the data first gets UTF-8 encoded and the `TERMINATOR`
+        gets appended.
+
+        Parameters
+        ----------
+        line : `str`
+            The line of data to be written.
```

Comment:
Please document the exceptions, including UnicodeError.

(If you want to defer the others to self.write, that's fine with me, though I don't know how practical it is to do that -- i.e. if it's possible to get sphinx to make that look reasonable).